File: stack/valid-parentheses/valid-parentheses.js
function isValid(str){
  let stack = [];
  let closeToOpen = new Map();

  closeToOpen.set(']','[')
  closeToOpen.set(')','(')
  closeToOpen.set('}','{')

  for(let i = 0; i < str.length; i++){
    if(closeToOpen.has(str[i])){
      if(
          stack.length>0 
          &&
          stack[stack.length-1] == closeToOpen.get(str[i])){
        stack.pop();
      }else{
        return false;
      }
    }else{
      stack.push(str[i]);
    }
  }
  return stack.length == 0
}

console.log(isValid('(())'))

File: stack/valid-parentheses/valid-parentheses.py
def isValid(s:str)->bool:
  close_to_open = {
    '}':'{',
    ')':'(',
    ']':'['
  }
  stack = []
  
  for c in s:
    if c in close_to_open:
      if stack and close_to_open[c] == stack[-1]:
        stack.pop()
      else:
        return False
    else:
      stack.append(c)
  return not stack
  
valid = isValid(")")
print(valid)